### PR TITLE
feat: add application metadata  …

### LINKED ISSUE #27: Create a fifth specification to separate out workload metadata
**Current Behavior**
The specification nominally considers for the description and all of the interesting metadata of the application (e.g. application name, application version, helm chart version #, and so on).

**Desired Behavior**
The definition of the workload profile definition should be expanded on. 

Consider definitions like CNAB, Docker Compose, Helm, and so on.

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #27

### CODE REVIEW

File: metadata.proto [hunk 1]
```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+package SMPS;
+
+option go_package = "spec;spec";
+
+message ApplicationMetadata {
+
+    enum application_protocol {
+        http = 0;
```
Comment (leecalcote) [inline metadata.proto:9]: Eventually, we'll want this to refer to the traffic.proto - https://github.com/layer5io/service-mesh-performance-specification/issues/38

And this, too - https://github.com/layer5io/service-mesh-performance-specification/blob/master/smps.proto#L13

Comment (kushthedude) [reply metadata.proto:9]: Makes sense, do you want me to do this in the same PR ?

Comment (leecalcote) [reply metadata.proto:9]: Thanks. No, we can circle back here.

Comment (leecalcote) [reply metadata.proto:9]: @mandarjog @carolynhu have you done any performance testing comparing efficiencies / handling of gRPC vs Nats requests?

Comment (carolynhu) [reply metadata.proto:9]: no, we did not compare those yet.

**Comment by kushthedude** [comment]
@leecalcote I initialised a prototool YAML for configuration right now it is just fixing the formatting, I will incorporate rules later in PR for the fix for #32

File: metadata.proto (hunk 1 above)
Comment (leecalcote) [reply metadata.proto]: Oh, I see. Thank you for confirming, @carolynhu.

File: metadata.proto [hunk 2]
```diff
@@ -1,21 +1,20 @@
 syntax = "proto3";
+
 package SMPS;
 
 option go_package = "spec;spec";
 
 message ApplicationMetadata {
-
-    enum application_protocol {
```
Comment (leecalcote) [inline metadata.proto]: Currently, we're not referring to OAM, but may want to do so in the future.

**Pull request merged**